### PR TITLE
Controller test

### PR DESCRIPTION
**Pull request by Raychenko88**
Controller test

### CODE REVIEW

File: src/test/java/info/sjd/controller/CartControllerTest.java [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+                login("test_login111").
+                password("test_pass").
+                firstName("test_fn").
+                lastName("test_ln").
+                build();
+        Cart cart = Cart.builder().
+                closed(0).user(user).
+                creationTime(currentTime).
+                build();
+        when(cartService.findById(anyInt())).thenReturn(cart);
```
Comment (Ignatenko2207) [inline src/test/java/info/sjd/controller/CartControllerTest.java:121]: You must mock delete method too. 
Like
doNothing().when(cartService)...

File: src/test/java/info/sjd/controller/CartControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+        RequestEntity<Cart> requestEntity = new RequestEntity<>(cart, HttpMethod.DELETE, new URI("/cart/3"));
+        ResponseEntity<Cart> responseEntity = testRestTemplate.exchange(requestEntity, Cart.class);
+        assertEquals(responseEntity.getStatusCode(), HttpStatus.OK);
+        verify(cartService, times(1)).findById(anyInt());
```
Comment (Ignatenko2207) [inline src/test/java/info/sjd/controller/CartControllerTest.java:126]: Check delete method invocation too

File: src/test/java/info/sjd/controller/ItemControllerTest.java [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+                .host("/item")
+                .path("all-available").build();
+        RequestEntity<Item> requestEntity = new RequestEntity<>(HttpMethod.GET, new URI(uriComponents.toString()));
+        ResponseEntity<List> responseEntity = testRestTemplate.exchange(requestEntity, List.class);
+        assertEquals(responseEntity.getStatusCode(), HttpStatus.OK);
+        verify(itemService, times(1)).getAllAvailable();
+    }
+
+    @Test
+    void deleteTest() throws URISyntaxException {
```
Comment (Ignatenko2207) [inline src/test/java/info/sjd/controller/ItemControllerTest.java:171]: You must mock delete method too.
Do not forget check its invocation.

File: src/test/java/info/sjd/controller/OrderControllerTest.java [hunk 4]
```diff
@@ -0,0 +1,323 @@
...
+                code("t_code").
+                price(123).
+                availability(1).
+                build();
+        Order order = Order.builder().
+                item(item).
+                cart(cart).
+                amount(10).
+                build();
+        when(orderService.findById(anyInt())).thenReturn(order);
```
Comment (Ignatenko2207) [inline src/test/java/info/sjd/controller/OrderControllerTest.java:135]: New object without fields initialization is enough. It's redundant to initialize other objects.
Also you must mock and check delete method

File: src/test/java/info/sjd/controller/UserControllerTest.java [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+                .host("/user")
+                .path("by-login-and-password").query("login={keyword}").query("password={keyword}").buildAndExpand("log", "pass");
+        RequestEntity<User> requestEntity = new RequestEntity<>(HttpMethod.GET, new URI(uriComponents.toString()));
+        ResponseEntity<User> responseEntity = testRestTemplate.exchange(requestEntity, User.class);
+        assertEquals(responseEntity.getStatusCode(), HttpStatus.OK);
+        verify(userService, times(1)).findByLoginAndPassword(anyString(), anyString());
+    }
+
+    @Test
+    void deleteTest() throws URISyntaxException {
```
Comment (Ignatenko2207) [inline src/test/java/info/sjd/controller/UserControllerTest.java:145]: The same situation as previous. You must mock and check delete method

File: src/test/java/info/sjd/controller/CartControllerTest.java (hunk 1 above)
Comment (Raychenko88) [reply src/test/java/info/sjd/controller/CartControllerTest.java:121]: done

File: src/test/java/info/sjd/controller/CartControllerTest.java (hunk 2 above)
Comment (Raychenko88) [reply src/test/java/info/sjd/controller/CartControllerTest.java:126]: done

File: src/test/java/info/sjd/controller/ItemControllerTest.java (hunk 3 above)
Comment (Raychenko88) [reply src/test/java/info/sjd/controller/ItemControllerTest.java:171]: done

File: src/test/java/info/sjd/controller/OrderControllerTest.java (hunk 4 above)
Comment (Raychenko88) [reply src/test/java/info/sjd/controller/OrderControllerTest.java:135]: done

File: src/test/java/info/sjd/controller/UserControllerTest.java (hunk 5 above)
Comment (Raychenko88) [reply src/test/java/info/sjd/controller/UserControllerTest.java:145]: done

**Review by Ignatenko2207 — APPROVED** [review]

**Pull request merged**